libraries/rush-lib/src/logic/operations/ProjectLogWritable.ts

```diff
@@ -51,102 +77,85 @@ export class ProjectLogWritable extends TerminalWritable {
-    options: {
-      enableChunkedOutput?: boolean;
-    } = {}
-  ) {
-    super();
-    const { enableChunkedOutput = false } = options;
-    this._terminal = terminal;
-
-    this._enableChunkedOutput = enableChunkedOutput;
-
-    // Delete the legacy logs
-    const { logPath: legacyLogPath, errorLogPath: legacyErrorLogPath } = ProjectLogWritable.getLogFilePaths({
-      project,
-      logFilenameIdentifier: 'build',
-      isLegacyLog: true
-    });
-    FileSystem.deleteFile(legacyLogPath);
-    FileSystem.deleteFile(legacyErrorLogPath);
-
-    const {
+    enableChunkedOutput: boolean,
+    {
       logPath,
       errorLogPath,
       logChunksPath,
       relativeLogPath,
       relativeErrorLogPath,
       relativeLogChunksPath
-    } = ProjectLogWritable.getLogFilePaths({
-      project,
-      logFilenameIdentifier
-    });
+    }: ILogFilePaths
+  ) {
+    super();
+    this._terminal = terminal;
+    this._enableChunkedOutput = enableChunkedOutput;
+
     this.logPath = logPath;
     this.errorLogPath = errorLogPath;
     this.logChunksPath = logChunksPath;
     this.relativeLogPath = relativeLogPath;
     this.relativeErrorLogPath = relativeErrorLogPath;
     this.relativeLogChunksPath = relativeLogChunksPath;
 
-    if (legacyLogPath !== this.logPath) {
-      FileSystem.deleteFile(this.logPath);
-    }
-
-    if (legacyErrorLogPath !== this.errorLogPath) {
-      FileSystem.deleteFile(this.errorLogPath);
-    }
-
-    this._logWriter = FileWriter.open(this.logPath);
-    this._chunkWriter = FileWriter.open(this.logChunksPath);
+    this._logWriter = FileWriter.open(logPath);
+    this._chunkWriter = FileWriter.open(logChunksPath);
   }
 
-  public static getLogFilePaths({
+  public static async initializeAsync({
     project,
+    terminal,
     logFilenameIdentifier,
-    isLegacyLog = false
-  }: {
-    project: RushConfigurationProject;
-    logFilenameIdentifier: string;
-    isLegacyLog?: boolean;
-  }): {
-    logPath: string;
-    logChunksPath: string;
-    errorLogPath: string;
-    relativeLogPath: string;
-    relativeErrorLogPath: string;
-    relativeLogChunksPath: string;
-  } {
-    const logFileBaseName: string = getRelativeLogFilePathBase(
+    enableChunkedOutput = false
+  }: IProjectLogWritableOptions): Promise<ProjectLogWritable> {
+    const logFilePaths: ILogFilePaths = ProjectLogWritable.getLogFilePaths({
       project,
-      logFilenameIdentifier,
-      isLegacyLog,
-      RushConstants.rushLogsFolderName
-    );
+      logFilenameIdentifier
+    });
 
-    const chunkLogFileBaseName: string = getRelativeLogFilePathBase(
-      project,
-      logFilenameIdentifier,
-      isLegacyLog,
-      `.rush/${RushConstants.rushTempFolderName}/operations`
-    );
+    const { logFolderPath, logChunksFolderPath, logPath, errorLogPath, logChunksPath } = logFilePaths;
+    await Promise.all([
+      FileSystem.ensureFolderAsync(logFolderPath),
+      FileSystem.ensureFolderAsync(logChunksFolderPath),
```

Comment:
Does this really need to be its own folder? It's just the same log in a different format.

Reply:
Moving them into the same folder will be in one of the upcoming refactor.s